app/src/query.rs

```diff
@@ -402,61 +417,95 @@ impl<'app, D: Dependencies<'app>> ApplicationReader<'app, D> {
         after: Option<&StructTag>,
         limit: u32,
     ) -> Result<Vec<StructTag>> {
-        let height = self.resolve_height(height)?;
-        Ok(self.state_queries.move_list_resources(
-            address.to_move_address(),
-            height,
-            after,
-            limit,
-        )?)
+        let hash = self.resolve_height(height)?.hash;
+        Ok(self
+            .state_queries
+            .move_list_resources(address.to_move_address(), hash, after, limit)?)
     }
 
-    fn resolve_height(&self, height: BlockNumberOrTag) -> Result<u64> {
+    fn resolve_height(&self, height: BlockNumberOrTag) -> Result<alloy::rpc::types::eth::Header> {
```

Comment:
Nit:
```suggestion
    fn resolve_height_to_header(&self, height: BlockNumberOrTag) -> Result<alloy::rpc::types::eth::Header> {
```
or something along these lines should sound clearer as I wouldn't expect getting a header back from just a height/tag as the most default thing.